### PR TITLE
add keybase authentication option in netconf provider

### PR DESCRIPTION
**Pull request by ylil93**
- add keybase authentication option in netconf provider
- update testing
- update documentation

### CODE REVIEW

File: sdk/cpp/core/docsgen/api/ydk/providers/netconf_provider.rst [hunk 1]
```diff
@@ -39,13 +39,55 @@ NetconfServiceProvider
...
+        int port = 830, \
+        bool on_demand = true, \
+        int timeout = -1)
+
+        Constructs an instance of ``NetconfServiceProvider`` connect to a server which **has** to support model download
+
+        :param repo: Reference to an instance of :cpp:class:`Repository<ydk::path::Repository>`
+        :param address: IP address of the device supporting a netconf interface.
+        :param username: Username to log in to the device.
+        :param private_key_path: Path to private key file.
```
Comment (abhikeshav) [inline sdk/cpp/core/docsgen/api/ydk/providers/netconf_provider.rst]: Can you indicate if this is the absolute path or relative path?

File: sdk/cpp/core/src/path/netconf_session.cpp [hunk 2]
```diff
@@ -100,10 +100,85 @@ NetconfSession::NetconfSession(const string& address,
...
+    initialize_client_with_key(address, username, private_key_path, public_key_path,
+        port, timeout);
+    auto caching_option = common_cache ? path::ModelCachingOption::COMMON : path::ModelCachingOption::PER_DEVICE;
+    path::Repository repo(caching_option);
+    initialize_repo(repo, on_demand);
+   YLOG_INFO("Connected to {} on port {} using SSH with timeout of {}",
+       address, port, timeout);
+}
+
+// NetconfSession::NetconfSession(
```
Comment (abhikeshav) [inline sdk/cpp/core/src/path/netconf_session.cpp]: Can the commented lines be removed?

File: sdk/cpp/tests/test_netconf_provider.cpp [hunk 3]
```diff
@@ -49,3 +49,22 @@ TEST_CASE("get_capabilities")
     NetconfServiceProvider provider{ "127.0.0.1", "admin", "admin", 12022};
     CHECK_NOTHROW(provider.get_capabilities());
 }
+
+// TEST_CASE("CreateKeybaseAuthRepo_Domingo")
```
Comment (abhikeshav) [inline sdk/cpp/tests/test_netconf_provider.cpp]: Can the commented lines be removed?

File: sdk/python/core/tests/test_netconf_provider.py [hunk 4]
```diff
@@ -53,6 +52,37 @@ def test_get_capabilities(self):
...
+        private = "%s/ssh_host_rsa_key" % dir_path
+        public = "%s/ssh_host_rsa_key.pub" % dir_path
+        provider = NetconfServiceProvider(
+            address = self.hostname,
+            username = self.username,
+            private_key_path = private,
+            public_key_path = public,
+            port = self.port)
+
+        # private = "%s/domingo_sshkey" % dir_path
```
Comment (abhikeshav) [inline sdk/python/core/tests/test_netconf_provider.py]: Can the commented lines be removed?

File: sdk/cpp/core/docsgen/api/ydk/providers/netconf_provider.rst (hunk 1 above)
Comment (ylil93) [reply sdk/cpp/core/docsgen/api/ydk/providers/netconf_provider.rst]: I've tried both and it takes either, but yeah I can specify that.

**Pull request merged**